### PR TITLE
DAOS-5043 control: dmg pool create fails if user but no group

### PR DESCRIPTION
**Pull request by tanabarr**
'dmg pool create ... -u=root' will fail because when passing a user
but no group, in formatUserGroups() the egid is only fetched if both
are empty before sending to the server.

- update to populate empty group or user parameter
- use external user wrapper in auth pkg to mock and add test coverage
- fix printed sizes in pool create to use base-10 units not base-2

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/lib/control/pool.go [hunk 1]
```diff
@@ -51,21 +51,26 @@ func checkUUID(uuidStr string) error {
 	return errors.Wrapf(err, "invalid UUID %q", uuidStr)
 }
 
-// formatNameGroup converts system names to principal and if both user and group
-// are unspecified, takes effective user name and that user's primary group.
+// formatNameGroup converts system names to principal and ensures both user and
+// group are, takes effective user name and that user's primary group.
```
Comment (kjacque) [inline src/control/lib/control/pool.go:55]: The second part of the comment got a bit mixed up. Maybe something like "formatNameGroup converts system names to principals. If user or group isn't provided, the effective user and/or effective group will be used."

**Comment by tanabarr** [comment]
intermittent failures in CI:
functional_hardware_large failed to provision wolf-142
functional_hardware_medium failed to install spdk-tools
CORCI-935
DAOS-623

**Comment by mjmac** [comment]
Passed all suites, just on different runs.

**Comment by tanabarr** [comment]
previous CI run: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-2816/3/pipeline (no code changes compared to the commit that the most recent run was based on) passed hardware small and large only failing medium due to a provisioning failure

**Pull request merged**